fairlearn/metrics/extra_metrics.py

```diff
@@ -0,0 +1,28 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+import sklearn.metrics as skm
+
+
+def specificity_score(y_true, y_pred, sample_weight=None):
+    # aka True Negative Rate
+    cm = skm.confusion_matrix(y_true, y_pred, sample_weight=sample_weight)
+    # Taken from
+    # https://scikit-learn.org/stable/modules/generated/sklearn.metrics.confusion_matrix.html
+    # This restricts us to binary classification
+    tn, fp, fn, tp = cm.ravel()
+    return tn / (tn+fp)
+
+
+def miss_score(y_true, y_pred, sample_weight=None):
+    # aka False Negative Rate
+    tpr = skm.recall_score(y_true, y_pred, sample_weight=sample_weight)
+
+    # FNR == 1 - TPR
+    return 1 - tpr
+
+
+def fallout_score(y_true, y_pred, sample_weight=None):
+    # aka False Positive Rate
+    # Since we use specificity, also restricted to binary classification
+    return 1 - specificity_score(y_true, y_pred, sample_weight)
```

Comment:
Would you consider including fpr, fnr? I can write them in the widget otherwise. Similarly, we need the outcomes = class 0 binned by group and overall.

Reply:
"miss rate" and "fallout rate" are not scores (score is something that you want to maximize). if anything, they are error metrics--but the names are somewhat obscure, so I'm not sure we want to define them. I would suggest just defining specificity_score as you do, and then metrics for fpr, fnr, tpr, tnr, etc.

Reply:
I can rename to `fallout_rate` per @MiroDudik 's comment, but... this is the false positive rate. I understood from @mesameki 's interviews that these were the names data scientists preferred?

In any case, this needs revision to cope with non-binary classification.